crates/sui-indexer/src/handlers/checkpoint_handler.rs

```diff
@@ -188,6 +231,62 @@ where
         }
     }
 
+    async fn start_checkpoint_commit(&self) -> Result<(), IndexerError> {
+        info!("Indexer checkpoint commit task started...");
+        loop {
```

Comment:
can we also measure the overall latency of each loop iteration(including how long does it take to wait for the next checkpoint). If this number is different than `persistence`, then we know it it still waiting for downloading and thus need further optimization

Reply:
actually I was wrong, from the charts we were looking at, looks like
```
self.metrics.fullnode_checkpoint_download_latency.start_timer();
```
will not reset the timer, thus the current download time includes "wait for new checkpoint" time, that's why this download time was even higher when indexer has caught up.

I will do 2 metrics here:
- one is wait and download
- the other is download

to double check, if "wait time" somehow becomes long, what's the optimization do you refer? the wait time should mostly just be "checkpoint generation interval"?

Reply:
actually, I did not figure a good way to "reset" timer on Prometheus timer, will do that later, b/c today fullnode_checkpoint_wait_and_download_latency can be used for both
- when indexer is catching up, it is the real downloading time without wait
- when indexer has caught up, it is the wait + download time

Reply:
what about something like
```
start = Instant::now();
while wait_for_checkpoint
  ....
delta = Instant::now() - start
prom.log(delta)
```

we can have a metric for waiting_for_checkpoint and download_checkpoint? i find myself more curious about the times individually than wait + download

Reply:
@wlmyng looks great, feel free to open a PR for that.